docker/release.sh

```diff
@@ -0,0 +1,55 @@
+#!/bin/bash
+# Usage:
+# // To build and publish the Docker image to the internal container registry.
+# $ DESTINATION=internal INTERNAL_CONTAINER_REGISTRY=<internal_registry> ./release.sh
+# // To publish the Docker image to the public container registry.
+# $ DESTINATION=public INTERNAL_CONTAINER_REGISTRY=<internal_registry> ./release.sh
+set -ex
+
+DOCKER_IMAGE_NAME="stackdriver-logging-agent"
+PUBLIC_CONTAINER_REGISTRY="gcr.io/stackdriver-agents"
+DESTINATION="${DESTINATION:-internal}"
+
+# Set version variables. See docker/VERSION file for details.
+source VERSION
+
+# Decide the Docker image version to be released based on the current Git branch.
+GIT_BRANCH=`git status | grep "On branch" | sed 's/On branch //g'`
+case "${GIT_BRANCH}" in
+  master)
+    DOCKER_IMAGE_VERSION="${DOCKERFILE_VERSION}-${GOOGLE_FLUENTD_VERSION}";
+    ;;
+  # TODO(qingling128): Remove this case when "kubernetes" branch is deprecated.
+  kubernetes)
+    DOCKER_IMAGE_VERSION="${DOCKERFILE_VERSION}-${GOOGLE_FLUENTD_VERSION}-k8s";
+    ;;
+  *)
+    echo "This script should only be used for restricted branches. Exiting.";
+    exit 1;
+    ;;
+esac
+
+
+if [[ -z "${INTERNAL_CONTAINER_REGISTRY}" ]]; then
```

Comment:
Renaming this to `STAGING_CONTAINER_REGISTRY` might be friendlier to OSS, that way users don't feel locked out, it's also the clearer intent behind what we're using it for.